src/sentry/monitors/tasks.py

```diff
@@ -129,26 +129,23 @@ def try_monitor_tasks_trigger(ts: datetime, partition: int):
     # timestamp. Use `int()` to keep the timestamp (score) as an int
     slowest_part_ts = int(slowest_partitions[0][1])
 
-    # TODO(epurkhiser): The `slowest_part_ts` is going to become the
-    # reference_ts for the rest of this function. But we don't want to flip
-    # this over quite yet since we want to make sure this is working as
-    # expected.
-
     precheck_last_ts = redis_client.get(MONITOR_TASKS_LAST_TRIGGERED_KEY)
     if precheck_last_ts is not None:
         precheck_last_ts = int(precheck_last_ts)
 
-    # If we have the same or an older reference timestamp from the most recent
-    # tick there is nothing to do, we've already handled this tick.
+    # If we have the same or an older timestamp from the most recent tick there
+    # is nothing to do, we've already handled this tick.
     #
-    # The scenario where the reference_ts is older is likely due to a partition
-    # being slightly behind another partition that we've already read from
-    if precheck_last_ts is not None and precheck_last_ts >= reference_ts:
+    # The scenario where the slowest_part_ts is older may happen when our
+    # MONITOR_TASKS_PARTITION_CLOCKS set did not know about every partition the
+    # topic is responsible for. Older check-ins may be processed after newer
+    # ones in diferent topics. This should only happen if redis loses state.
+    if precheck_last_ts is not None and precheck_last_ts >= slowest_part_ts:
```

Comment:
The case where the reference time (now `slowest_part_time`) is BEFORE the last time has changed. The comment now describes when this can happen.

The previous case was basically a symptom of partition desynchronization.

Reply:
Ah yeah good catch. I'll update the PR separately